docker/mnt/run.sh

```diff
@@ -0,0 +1,121 @@
+#! /bin/bash
+set -e
+
+if [ -z "$1" ]; then
+    # the main container need not stay running for services
+    exit 0
+fi
+
+function setup() {
+    [ -n "$1" ] && TEST="$1"
+
+    if [[ "$TEST" =~ ^python ]]; then
+
+        scripts/uninstall-requirements.sh
+        pip install \
+            -r requirements/requirements.txt \
+            -r requirements/dev-requirements.txt \
+            coveralls
+
+        # some kind of optimization?
+        # skip Running setup.py bdist_wheel for ... ?
+        rm -rf /root/.cache/pip
+
+        /usr/lib/jvm/jdk1.7.0/bin/keytool -genkey \
+            -keyalg RSA \
+            -keysize 2048 \
+            -validity 10000 \
+            -alias javarosakey \
+            -keypass onetwothreefourfive \
+            -keystore InsecureTestingKeyStore \
+            -storepass onetwothreefourfive \
+            -dname 'CN=Foo, OU=Bar, O=Bizzle, L=Bazzle, ST=Bingle, C=US'
+    fi
+
+    if [ "$TEST" = "javascript" -o "$JS_SETUP" = "yes" ]; then
+        npm install
+        bower install --config.interactive=false
+    fi
+
+    /mnt/wait.sh
+
+    if [[ "$TEST" =~ ^python ]]; then
+        su cchq -c "./manage.py create_kafka_topics"
+    fi
+}
+
+function run_tests() {
+    TEST="$1"
+    if [ "$TEST" != "javascript" -a "$TEST" != "python" -a "$TEST" != "python-sharded" ]; then
+        echo "Unknown test suite: $TEST"
+        exit 1
+    fi
+    shift
+    setup $TEST
+    su cchq -c "../run_tests $TEST $(printf " %q" "$@")"
```

Comment:
Yes, that's all correct.

The `_run_tests` thing is a hack that I don't like, but thought it was a good idea to run all non-setup code as a user other than root since the container is "[privileged](https://github.com/dimagi/commcare-hq/blob/943d45ddffe0d547d6b7defd355afd43105251a6/docker/hq-compose-bootstrap.yml#L6)" (which probably means it is root to the host OS).

`/mnt/lib` contains three directories: `node_modules`, `sharedfiles`, and `overlay`.

`overlay` is a read/write filesystem overlay for `/mnt/commcare-hq-ro`. An AUFS layered filesystem composed of `/mnt/lib/overlay` and `/mnt/commcare-hq-ro` is mounted at `/mnt/commcare-hq`. This allows the container to safely create and overwrite files in the repo without affecting anything outside the container.